aadwarf64/aadwarf64.rst

```diff
@@ -574,25 +582,33 @@ a CIE augmentation string.
 Call frame instructions
 -----------------------
 
-This ABI defines one vendor call frame instruction
-``DW_CFA_AARCH64_negate_ra_state``.
+This ABI defines two vendor call frame instructions:
+``DW_CFA_AARCH64_negate_ra_state`` and ``DW_CFA_AARCH64_negate_ra_state_with_pc``.
 
 .. class:: aadwarf64-vendor-cfa-operations
 
 .. table:: AArch64 vendor CFA operations
 
-   +------------------------------------+-------------+------------+-----------+-----------+
-   | Instruction                        | High 2 bits | Low 6 bits | Operand 1 | Operand 2 |
-   +====================================+=============+============+===========+===========+
-   | ``DW_CFA_AARCH64_negate_ra_state`` | 0           | ``0x2D``   | \-        | \-        |
-   +------------------------------------+-------------+------------+-----------+-----------+
+   +--------------------------------------------+-------------+------------+-----------+-----------+
+   | Instruction                                | High 2 bits | Low 6 bits | Operand 1 | Operand 2 |
+   +============================================+=============+============+===========+===========+
+   | ``DW_CFA_AARCH64_negate_ra_state``         | 0           | ``0x2D``   | \-        | \-        |
+   +--------------------------------------------+-------------+------------+-----------+-----------+
+   | ``DW_CFA_AARCH64_negate_ra_state_with_pc`` | 0           | ``0x2C``   | \-        | \-        |
+   +--------------------------------------------+-------------+------------+-----------+-----------+
 
 The ``DW_CFA_AARCH64_negate_ra_state`` operation negates bit[0] of the
 RA_SIGN_STATE pseudo-register. It does not take any operands.
-The ``DW_CFA_AARCH64_negate_ra_state`` must not be mixed with other DWARF
-Register Rule Instructions (GDWARF_, §6.4.2.3) on the RA_SIGN_STATE
-pseudo-register in one Common Information Entry (CIE) and Frame Descriptor
-Entry (FDE) program sequence.
+
+The ``DW_CFA_AARCH64_negate_ra_state_with_pc`` operation negates bit[0] and
+bit[1] of the RA_SIGN_STATE pseudo-register, and instructs the unwinder capture
+the current code location. The code location information can be used for
```

Comment:
This only required when it set to 1.
Location of the `DW_CFA_AARCH64_negate_ra_state` is flexible until the code doesn't change the SP/LR, but the `DW_CFA_AARCH64_negate_ra_state_with_pc` must be exact, I'd note this somewhere in the text.

> instructs the unwinder capture the current code location.

sounds like we need a register for this to store this frame info.

Reply:
I've added a paragraph describing the requirement for the exact location of `DW_CFA_AARCH64_negate_ra_state_with_pc` instructions.